runtime/src/main/java/org/corfudb/runtime/collections/SMRMap.java

```diff
@@ -31,7 +32,7 @@
 @SuppressWarnings("checkstyle:abbreviation")
 public class SMRMap<K, V>
         extends HashMap<K, V>
-        implements ISMRMap<K,V>, ICorfuSMR<SMRMap<K, V>> {
+        implements StreamingMap<K, V>, ISMRMap<K,V>, ICorfuSMR<SMRMap<K, V>> {
```

Comment:
The type relationships is already complex.

I think it would be cleaner to have ISMRMap extend ICorfuSMR. How can an object be SMR if it doesn't implement ICorfuSMR, all SMRObject types should extend/implement that type.

Reply:
I was trying to avoid the diamond problem that arises due to `AutoClosable` interface. One solution would be to force all concrete implementation of `ICorfuSMR` to implement `close()`, but due those tricky annotation we have (`@PassThrough @DontInstrument`), I am hesitant to go down that path.